core/ui/htmlcontent/src/main/java/social/firefly/core/ui/htmlcontent/HtmlStringExtensions.kt

```diff
@@ -71,3 +71,38 @@ fun String.htmlToSpannable(
             }
         }
 }
+
+@Suppress("MaxLineLength")
+/**
+ * replaces link html link spans of class "u-url mention" with the full user handle
+ *
+ * For example, it changes this string:
+ *
+ * <p>test <span class="h-card" translate="no"><a href="https://test.social/@obez" class="u-url mention">@<span>obez</span></a></span></p>
```

Comment:
I would think we'd want to show whatever the span text shows and link to whatever the link is, and not necessarily reformat it

Reply:
The problem is there is no way to insert this span into plain text (as far as I'm aware).  So the link is lost when we parse the html.  If your original post is:

```
hello @obez@test.social
```

it gets reformatted into:

`hello <a href="https://test.social/@obez" class="u-url mention">@<span>obez</span></a>`

So then when we try to edit that, the text in the edit post screen becomes

```
hello @obez
```

When that gets updated, it then assumes `@obez` is on your server, instead of on `test.social`, so the link breaks or points to the wrong person.

Mentions should always have the `u-url mention` class, and I'm only replacing mention links.  Any other link span should stay the same.

Reply:
You were right, it's just a bug in the server I was using to test.  I'm removed the usage of this function.  I think I'm gonna just keep the function here though, even though it's not used.  Just in case it comes in handy in the future.  It took me a while to figure it out